### PR TITLE
Review folder structure, file documentation and pragmas

### PR DESCRIPTION
**Pull request by KtorZ**
# Issue Number

#20

# Overview

- Review haddock documentation for Cardano.Wallet.Binary
- Remove unnecessart intermediary modules 'BinaryHelpers' and 'Binary/Helpers'
- Rename and organize test data to reflect source code organization
- Add license and copyright to source files' header

# Comments

I've split the previous PR (#34 into two, and kept #34 really just about the tx id)

### CODE REVIEW

File: src/Cardano/Wallet/Binary.hs [hunk 1]
```diff
@@ -1,36 +1,28 @@
...
--- | These are (partial) CBOR decoders for blocks and block headers.
--- Note that we ignore most of the block's and header's content and only
--- retrieve the pieces of information relevant to us, wallet (we do assume a
--- trusted node and therefore, we needn't to care about verifying signatures and
--- blocks themselves).
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: MIT
+--
+-- These are (partial) CBOR decoders for blocks and block headers. Note that we
```
Comment (Anviking) [inline src/Cardano/Wallet/Binary.hs:8]: I know this is not part of the PR, but is the `CBOR` encoding specific to the rust bridge?

If it is specific, maybe the fact could permeate the folder structure, and/or be mentioned here.

If it is not specific, that would still be great to mention.

File: src/Cardano/Wallet/Binary.hs [hunk 2]
```diff
@@ -376,3 +385,49 @@ decodeUpdateProof :: CBOR.Decoder s ()
 decodeUpdateProof = do
     _ <- CBOR.decodeBytes -- Update Hash
     return ()
+
+
+-- * Helpers
```
Comment (Anviking) [inline src/Cardano/Wallet/Binary.hs:394]: 👍

File: src/Cardano/Wallet/Binary.hs [hunk 3]
```diff
@@ -1,36 +1,28 @@
...
 {-# LANGUAGE RankNTypes #-}
-{-# OPTIONS_GHC -fno-warn-name-shadowing #-}
 
--- | These are (partial) CBOR decoders for blocks and block headers.
--- Note that we ignore most of the block's and header's content and only
--- retrieve the pieces of information relevant to us, wallet (we do assume a
--- trusted node and therefore, we needn't to care about verifying signatures and
--- blocks themselves).
+-- |
+-- Copyright: © 2018-2019 IOHK
```
Comment (Anviking) [inline src/Cardano/Wallet/Binary.hs:5]: I assume this is good, and for when an individual file is included in another project.

Other IOHK projects doesn't seem to have this. Any particular reason for why we/now?

File: src/Cardano/Wallet/Binary.hs [hunk 4]
```diff
@@ -69,14 +75,15 @@ decodeAttributes = do
...
+{-# ANN decodeBlock ("HLint: ignore Use <$>" :: String) #-}
 decodeBlock :: CBOR.Decoder s Block
 decodeBlock = do
     CBOR.decodeListLenCanonicalOf 2
     t <- CBOR.decodeWordCanonical
     case t of
         0 -> do -- Genesis Block
             _ <- CBOR.decodeListLenCanonicalOf 3
-            header <- decodeGenesisBlockHeader
+            h <- decodeGenesisBlockHeader
```
Comment (Anviking) [inline src/Cardano/Wallet/Binary.hs:90]: 🤔 sure…

File: src/Cardano/Wallet/Binary.hs [hunk 5]
```diff
@@ -343,16 +351,17 @@ decodeTxIn = do
     decodeTxIn' :: CBOR.Decoder s TxIn
     decodeTxIn' = do
         _ <- CBOR.decodeListLenCanonicalOf 2
-        txId <- CBOR.decodeBytes
+        tx <- CBOR.decodeBytes
         index <- CBOR.decodeWord32
-        return $ TxIn (Hash txId) index
```
Comment (Anviking) [inline src/Cardano/Wallet/Binary.hs:348]: 🤔 I guess the reasoning here is that the `id` part of `txId` is obvious.

I think I'd prefer
```haskell
tx <- Hash <$> CBOR.decodeBytes
...
return $ TxIn tx index
```

to make it obvious that we're not hashing a tx, but that `tx` is a `Hash`.

File: test/unit/Cardano/Wallet/Binary/PackfileSpec.hs [hunk 6]
```diff
@@ -34,7 +34,7 @@ testTwoBlobs = packFileHeader
 -- Get this file from cardano-http-bridge with:
 -- wget -O test/data/epoch-mainnet-104 http://localhost:8080/mainnet/epoch/104
 testPackfile :: FilePath
-testPackfile = "test/data/epoch-mainnet-104"
+testPackfile = "test/data/Cardano/Wallet/Binary/PackfileSpec-epoch-mainnet-104"
```
Comment (Anviking) [inline test/unit/Cardano/Wallet/Binary/PackfileSpec.hs:37]: 👍  I'm starting to wonder whether there are editor plugins for browsing the shared structure

**Review by Anviking — APPROVED** [review]
Nice 👍 

One concrete naming-suggestion, and a some non-important questions.

Also, looks like you both moved and modified code-sections in the same commit. I find it practical to split up into two commits:
1. Move / reorder
2. Modify

for readability.

Would't have minded one commit per bullet-point either. Here it would have accomplished the reorder-modify-separation at the same time.

File: src/Cardano/Wallet/Binary.hs (hunk 1 above)
Comment (KtorZ) [reply src/Cardano/Wallet/Binary.hs:8]: This isn't specific to the http-bridge. That format is used across the whole Cardano ecosystem and will be use, at least in the first release, of the Shelley era.

File: src/Cardano/Wallet/Binary.hs (hunk 3 above)
Comment (KtorZ) [reply src/Cardano/Wallet/Binary.hs:5]: This is actually a standard practice in Haskell modules that are exported. This get parsed by `haddock` to generate a header for each module.

image

File: src/Cardano/Wallet/Binary.hs (hunk 4 above)
Comment (KtorZ) [reply src/Cardano/Wallet/Binary.hs:90]: Yeah, there was a hlint annotation at the root of the file asking to "ignore name shadowing", I removed it and adjusted a few names to get rid of the warnings.

File: test/unit/Cardano/Wallet/Binary/PackfileSpec.hs (hunk 6 above)
Comment (KtorZ) [reply test/unit/Cardano/Wallet/Binary/PackfileSpec.hs:37]: :man_shrugging:

**Comment by KtorZ** [comment]
> looks like you both moved and modified code-sections in the same commit. I find it practical to split up into two commits

@Anviking ^^^ Yeah, it was the case originally and when I did the split of this PR from #34, I did squash everything in one commit. I don't know why :/ ... Your remark is very sensible though -- sorry about that.

File: src/Cardano/Wallet/Binary.hs (hunk 1 above)
Comment (KtorZ) [reply src/Cardano/Wallet/Binary.hs:8]: Added a small disclaimer in the file's header:

https://github.com/input-output-hk/cardano-wallet/compare/ebe31ebbfc96a055b44ae3da687aa6eef6b939d5..7a6593d11547b4be6abd5752dc4cea96f002a507

File: src/Cardano/Wallet/Binary.hs (hunk 5 above)
Comment (KtorZ) [reply src/Cardano/Wallet/Binary.hs:348]: Done ( https://github.com/input-output-hk/cardano-wallet/compare/ebe31ebbfc96a055b44ae3da687aa6eef6b939d5..7a6593d11547b4be6abd5752dc4cea96f002a507 )

**Pull request merged**